File: daily/jan1/solution.go
package main

import (
	"fmt"
	"time"
)

// Solution represents the solution structure
type Solution struct{}

func (s *Solution) PlusOne(digits []int) []int {
	carry := 1
	result := make([]int, 0, len(digits)+1)

	for i := len(digits) - 1; i >= 0; i-- {
		sum := digits[i] + carry
		result = append([]int{sum % 10}, result...)
		carry = sum / 10
	}

	if carry > 0 {
		result = append([]int{carry}, result...)
	}

	return result
}

func main() {
	sol := &Solution{}
	digits := []int{1, 2, 3}

	// Start timing
	start := time.Now()
	result := sol.PlusOne(digits)
	duration := time.Since(start)

	// Print result
	fmt.Println(result)

	// Print execution time
	fmt.Printf("Execution time: %.6f ms\n", float64(duration.Nanoseconds())/1e6)
}

File: daily/jan1/solution.cpp
#include <chrono>
#include <iomanip>
#include <iostream>
#include <vector>

using namespace std;
using namespace std::chrono;

class Solution {
public:
  vector<int> plusOne(vector<int> &digits) {
    vector<int> result;
    int carry = 1;

    for (int i = digits.size() - 1; i >= 0; --i) {
      int sum = digits[i] + carry;
      result.insert(result.begin(), sum % 10);
      carry = sum / 10;
    }

    if (carry > 0) {
      result.insert(result.begin(), carry);
    }

    return result;
  }
};

int main() {
  Solution sol;
  vector<int> digits = {1, 2, 3};

  // Start timing
  auto start = high_resolution_clock::now();
  vector<int> result = sol.plusOne(digits);
  auto end = high_resolution_clock::now();

  // Calculate duration
  auto duration = duration_cast<nanoseconds>(end - start);

  // Print result
  cout << "[";
  for (size_t i = 0; i < result.size(); ++i) {
    cout << result[i];
    if (i < result.size() - 1)
      cout << ", ";
  }
  cout << "]" << endl;

  // Print execution time
  cout << fixed << setprecision(6);
  cout << "Execution time: " << duration.count() / 1e6 << " ms" << endl;

  return 0;
}
